### ISSUE TITLE
Pawn, related to Havoc

### ISSUE THREAD

**Issue by jindalpriyanshu101** [Mellnik/havoc#34]
Hi Mellnik i hope you are fine, from past 5-6days i was trying to compile main.pwn to amx of build 38 but am unable to do this and having this error, have not touched the script yet too so i think i miss a include maybe can you help me to know how to fix this one... also am using all the exact versions mentioned in script of plugins and includes (not all)... as i know that too that NEF clearly said there is no support for any of these GMs but please may i have a solution for this because i really want to reopen havoc (i know we are not allowed to open any gm with NEF terms, but i think i should do it.. also give all the credits to NEF team only, i myself not going to take any credits) Here's the errors which i got during compiling...

> F:\havoc-master\pawno\include\YSI\..\YSI_Visual\y_commands/impl.inc(228) : error 025: function heading differs from prototype
F:\havoc-master\pawno\include\YSI\..\YSI_Visual\y_commands/impl.inc(229) : error 025: function heading differs from prototype
F:\havoc-master\gamemodes\R38\main.pwn(3916) : error 025: function heading differs from prototype
F:\havoc-master\gamemodes\R38\main.pwn(3988) : error 025: function heading differs from prototype
F:\havoc-master\gamemodes\R38\main.pwn(7882) : warning 217: loose indentation
F:\havoc-master\gamemodes\R38\main.pwn(30220) : warning 203: symbol is never used: "boost"
F:\havoc-master\gamemodes\R38\main.pwn(30258) : warning 203: symbol is never used: "boost"
F:\havoc-master\gamemodes\R38\main.pwn(31494) : warning 203: symbol is never used: "MessageText"
F:\havoc-master\gamemodes\R38\main.pwn(31494) : warning 203: symbol is never used: "NPCLabelHandle"
F:\havoc-master\gamemodes\R38\main.pwn(31494) : warning 203: symbol is never used: "g_EventVehicle"
Pawn compiler 3.2.3664	 	 	Copyright (c) 1997-2006, ITB CompuPhase

4 Errors.

i hope you respond to this soon :)

**Comment by Mellnik** [comment]
Hi. There are no terms for you to not open a server. This is OSS published under the Apache License 2.0.

As for your errors: It looks like your y_commands include does not match the one used in the script. So try to find an older version or update the script.

**Comment by jindalpriyanshu101** [comment]
hello sir, sir as you said to try any older version of YSI i tried every possible way but unable to found YSI v3 or lower :( also samp forums got removed about a weak ago :| can you pls provide me your files that you used 😓 am sure will not give anyone will keep it upto me... but as it was clearly written no support will be given so i don't think you gonna give but still its my pleasure to have those files sir... i will host srv for myself only will not public it for sure if you give sir 😓

**Comment by Mellnik** [comment]
[include.zip](https://github.com/Mellnik/havoc/files/5343603/include.zip)
Check this include folder. I just found them but not sure if those are the correct ones :)

**Comment by jindalpriyanshu101** [comment]
Hi sir, yes they were not correct BUT has fixed the previous errors and gave new ones xD but i guess there is a file which isn't in required version this is the error i got after compiling from merging these includes and mine which i found also i had taken care that not to replace any of your given include so after this i got:

**Compiler output**

> 

> F:\havoc-master\Gamemodes\R38\main.pwn(4066) : error 025: function heading differs from prototype
F:\havoc-master\Gamemodes\R38\main.pwn(4076) : error 025: function heading differs from prototype
F:\havoc-master\Gamemodes\R38\main.pwn(4765) : error 025: function heading differs from prototype
F:\havoc-master\Gamemodes\R38\main.pwn(7882) : warning 217: loose indentation
F:\havoc-master\Gamemodes\R38\main.pwn(30220) : warning 203: symbol is never used: "boost"
F:\havoc-master\Gamemodes\R38\main.pwn(30258) : warning 203: symbol is never used: "boost"
F:\havoc-master\Gamemodes\R38\main.pwn(31494) : warning 203: symbol is never used: "MessageText"
F:\havoc-master\Gamemodes\R38\main.pwn(31494) : warning 203: symbol is never used: "NPCLabelHandle"
F:\havoc-master\Gamemodes\R38\main.pwn(31494) : warning 203: symbol is never used: "g_EventVehicle"
Pawn compiler 3.2.3664	 	 	Copyright (c) 1997-2006, ITB CompuPhase


3 Errors.

**Here Lines which are different from prototype:**

**4066:**`public OnPlayerTakeDamage(playerid, issuerid, Float:amount, weaponid, bodypart)`
**4076:**`public OnPlayerGiveDamage(playerid, damagedid, Float:amount, weaponid, bodypart)`
**4765:**`public OnUnoccupiedVehicleUpdate(vehicleid, playerid, passenger_seat, Float:new_x, Float:new_y, Float:new_z, Float:vel_x, Float:vel_y, Float:vel_z)`

Pls let me know from which file/include they are managed i will try my best to find required version of it :)
Thanks for your all these support you gave to me :)

**Comment by Mellnik** [comment]
Read what is says. The function differs from the one defined in a_samp.
So update the code to match it :)

**Comment by jindalpriyanshu101** [comment]
Yeah i did i tried some versions of a_samp which i had instead of updating code coz i think i have not touched script yet so probably am using wrong version of includes or has missed an include, however when i used another(maybe the one with whom it has no problem with function heading) it rolls out the error of undefined for "CreateActor" wheres i have a_actor.inc _(also tried the latest version but still got same output so replaced with the one i was using for this script)_ present in folder i dont know what to do, am lost at this point... :| also used your given includes and after all that got this..

**The Output when i used correct a_samp.inc (i think)**

> F:\havoc-master\Gamemodes\R38\main.pwn(2924) : error 017: undefined symbol "CreateActor"
F:\havoc-master\Gamemodes\R38\main.pwn(2925) : error 017: undefined symbol "CreateActor"
F:\havoc-master\Gamemodes\R38\main.pwn(2926) : error 017: undefined symbol "CreateActor"
F:\havoc-master\Gamemodes\R38\main.pwn(2927) : error 017: undefined symbol "CreateActor"
F:\havoc-master\Gamemodes\R38\main.pwn(2928) : error 017: undefined symbol "CreateActor"
F:\havoc-master\Gamemodes\R38\main.pwn(2929) : error 017: undefined symbol "CreateActor"
F:\havoc-master\Gamemodes\R38\main.pwn(2930) : error 017: undefined symbol "CreateActor"
F:\havoc-master\Gamemodes\R38\main.pwn(2931) : error 017: undefined symbol "CreateActor"
F:\havoc-master\Gamemodes\R38\main.pwn(2932) : error 017: undefined symbol "SetActorInvulnerable"
F:\havoc-master\Gamemodes\R38\main.pwn(2933) : error 017: undefined symbol "SetActorInvulnerable"
F:\havoc-master\Gamemodes\R38\main.pwn(2934) : error 017: undefined symbol "SetActorInvulnerable"
F:\havoc-master\Gamemodes\R38\main.pwn(2935) : error 017: undefined symbol "SetActorInvulnerable"
F:\havoc-master\Gamemodes\R38\main.pwn(2936) : error 017: undefined symbol "SetActorInvulnerable"
F:\havoc-master\Gamemodes\R38\main.pwn(2937) : error 017: undefined symbol "SetActorInvulnerable"
F:\havoc-master\Gamemodes\R38\main.pwn(2938) : error 017: undefined symbol "SetActorInvulnerable"
F:\havoc-master\Gamemodes\R38\main.pwn(2939) : error 017: undefined symbol "SetActorInvulnerable"
F:\havoc-master\Gamemodes\R38\main.pwn(2940) : error 017: undefined symbol "ApplyActorAnimation"
F:\havoc-master\Gamemodes\R38\main.pwn(2941) : error 017: undefined symbol "ApplyActorAnimation"
F:\havoc-master\Gamemodes\R38\main.pwn(2942) : error 017: undefined symbol "ApplyActorAnimation"
F:\havoc-master\Gamemodes\R38\main.pwn(2943) : error 017: undefined symbol "ApplyActorAnimation"
F:\havoc-master\Gamemodes\R38\main.pwn(3558) : warning 235: public function lacks forward declaration (symbol "OnIncomingConnection")
F:\havoc-master\Gamemodes\R38\main.pwn(3576) : error 017: undefined symbol "BlockIpAddress"
F:\havoc-master\Gamemodes\R38\main.pwn(4066) : error 025: function heading differs from prototype
F:\havoc-master\Gamemodes\R38\main.pwn(4076) : error 025: function heading differs from prototype
F:\havoc-master\Gamemodes\R38\main.pwn(4154) : warning 235: public function lacks forward declaration (symbol "OnPlayerWeaponShot")
F:\havoc-master\Gamemodes\R38\main.pwn(4157) : error 017: undefined symbol "BULLET_HIT_TYPE_NONE"
F:\havoc-master\Gamemodes\R38\main.pwn(4173) : error 017: undefined symbol "BULLET_HIT_TYPE_PLAYER"
F:\havoc-master\Gamemodes\R38\main.pwn(4193) : error 017: undefined symbol "BULLET_HIT_TYPE_PLAYER"

Compilation aborted.Pawn compiler 3.2.3664	 	 	Copyright (c) 1997-2006, ITB CompuPhase


26 Errors.

Actually using another version gives more error instead of getting a fix of function heading...

EDIT - gave more info

**Comment by Mellnik** [comment]
You will need the include files which have CreateActor etc. So 0.3.7 right?

**Comment by jindalpriyanshu101** [comment]
Yes i think so,
Yes 0.3.7
but CreateActor came from a_actor na?
